packages/nimble-components/src/spinner/specs/README.md

```diff
@@ -0,0 +1,138 @@
+![SystemLink Material spinner](SlMatSpinner.gif)
+
+A variant of this spinner design is already implemented in the Web Config tool written in Blazor:  
+![Blazor Web Config spinner](BlazorWebConfigSpinner.gif)
+
+## Design
+
+The `nimble-spinner` is an animating indicator that can be placed in a particular region of a page to represent loading progress, or an ongoing operation, of an indeterminate / unknown duration.
+
+### API
+
+Component Name: `nimble-spinner`
+
+Properties/Attributes:
+
+-   `size`
+    -   String attribute controlling size of the component
+    -   Backed by an enum `SpinnerSize` with values `small` (16x16), `medium` (32x32), `large` (64x64)
+    -   If omitted, the default is `medium` (32x32)
+-   `appearance-variant`
+    -   default (undefined): Standard appearance (light/dark/color themes)
+    -   `alternate`: Alternate appearance/color, for the color theme only
+    -   Backed by enum `SpinnerAppearanceVariant`
+
+Methods, Events, CSS Classes, CSS Custom Properties: None
+
+### Anatomy
+
+Contains 2 `div` elements that are animated via CSS animations.
+
+Slot Names, Host Classes, Slotted Content/ Slotted Classes, CSS Parts: None
+
+### Angular integration
+
+Directive `NimbleSpinnerDirective` targeting selector `nimble-spinner`, with bindings for `size` and `appearance-variant`.
+
+### Blazor integration
+
+Standard Blazor implementation (`NimbleSpinner` deriving from `ComponentBase`), with bindings for `size` and `appearance-variant`.
+
+### Visual Appearance
+
+[Visual Design spec - Adobe XD](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/dece308f-79e7-48ec-ab41-011f3376b49b/)
+
+Light / dark themes only have a single appearance variant.  
+The color theme has 2 appearance variants - the "Color UI Alt." one from XD can be accessed with `appearance-variant="alternate"`.
+
+![New spinner design](NewSpinnerDesign.gif)
+
+## Implementation
+
+Component implementation will derive from the FAST `FoundationElement`.
+
+**Alternatives Considered:** We could derive from `fast-progress-ring` (radial progress indicator, which supports an indeterminate progress display), however then our component API would also pick up the `min`/`max`/`value` properties for determinate progress (and `paused` to pause the indicator), which we don't plan on supporting at this time. Additionally, the visual appearance of our spinner will be a little different from a radial progress bar, and we still may end up supporting a standard radial progress bar in the future.  
+For reference, the FAST spec for `progress` / `progress-ring` is [here](https://github.com/microsoft/fast/blob/802443ffb2b19a078f9b48f62e6d1a35e3276fb5/packages/web-components/fast-foundation/src/progress/README.md).
+
+### States
+
+Only one component state: visible and animating. Pausing not supported; spinner is expected to be added/removed from the DOM as needed.
+
+### Accessibility
```

Comment:
Thoughts on `prefers-reduced-motion`? Based on our [current interpretation of WCAG guidance](https://github.com/ni/nimble/blob/95cbaea9c0ae16b804816ac68204253735bc7b84/packages/nimble-components/CONTRIBUTING.md?plain=1) it seems like we'd want to disable animations for this component but that obviously makes it pretty much useless. Another option would be using a different animation with more fading and less motion. If we can find examples of how other design systems/or OSes handle this case, that would be helpful.

Reply:
Good question. I agree that turning off animation for the component entirely doesn't really feel right, since it's supposed to convey that an operation is happening.

Maybe turning off the sliding/easing on the animation, but letting the bits move still? Example:
![SpinnerNoSlideAnimation](https://user-images.githubusercontent.com/20709258/204910142-1dac6493-fd84-4b9c-a71f-ae97fec0bfbb.gif)

Reply:
I don't have any of the conditions that `prefers-reduced-motion` is meant to help with, but to me the un-animated version is even more jarring! Maybe try opacity fading between the states?

Reply:
Yea that's not very comfortable for me to look at lol
I'm not sure opacity between the states would look right to me either. Maybe something completely different should be shown like this fading in and out `...` https://github.com/alphagov/govuk-design-system-backlog/issues/164
Seems like a more univeral sign for "loading" or "waiting".

Reply:
@NIbokeefe, do you have any thoughts on a design to support this situation?

Reply:
Potential option:
Building on what @jattasNI suggested, "breathe" fade between two states.

Maybe these can get us close:
https://vmar76.medium.com/using-css-animations-to-visualize-breathing-techniques-7a20ee0aed5a
https://codepen.io/seansean11/pen/DgGYPY
https://codepen.io/saigowthamr/pen/PowdZqy

Reply:
Simple opacity fading between the 2 states:
![SpinnerFade1](https://user-images.githubusercontent.com/20709258/204957888-6b0c1bd3-5240-4918-bea8-4d2ae71b34ff.gif)
2x animation duration:
![SpinnerFade1HalfSpeed](https://user-images.githubusercontent.com/20709258/204957959-12277ece-5156-4512-a904-26c141e081b7.gif)

Reply:
Both of those hurt my eyes.

Reply:
> Good question. I agree that turning off animation for the component entirely doesn't really feel right, since it's supposed to convey that an operation is happening.
> 
> Maybe turning off the sliding/easing on the animation, but letting the bits move still? Example: ![SpinnerNoSlideAnimation](https://user-images.githubusercontent.com/20709258/204910142-1dac6493-fd84-4b9c-a71f-ae97fec0bfbb.gif)

This execution seems to be the best solution so far.  Moving the boxes without the transparent blur. 

Are we able to add 1 more mid frame where the box is half way?  So in your example there are 3 frames (8 total to complete the entire animation).

For your example: The box, starts on the left > then appears in the middle > then appears on the right.  If we dial in the timing on this it wouldn't be as jarring as two frames.

Reply:
Taking a shot. Haven't got the timing right just yet but:
https://typescript-ulw6zh.stackblitz.io

Reply:
Updated the spec to have an open issue for the `prefers-reduced-motion` appearance. I'll add another comment here in a bit with a link to a Storybook with some options.

Reply:
[Spinner storybook page with prefers-reduced-motion options to try out](https://60e89457a987cf003efc0a5b-vhzrflhysg.chromatic.com/?path=/story/spinner--spinner&args=size:large;forceReducedMotion:true)
It has a toggle to force the reduced motion variants to be on (so you don't have to go enable that in your OS settings first).

It has a few options:

- `fade`: Cross-fade between the 2 alternating bit positions
   - Original variant, we already eliminated this I think
- `middle-step`: Add 1 intermediate step between each position (still progresses clockwise)
   - Brandon's suggestion (hopefully I implemented it right)
- `fade-first`: Fade in and out: 1st position only
- `fade-both`: Fade in and out: (1st position, 2nd, [repeat])
   - [Nathan's pulse example](https://typescript-ulw6zh.stackblitz.io/) is basically a variation on this one, just with different fade/animation durations

I think I'm also leaning towards some form of the last one now (`fade-both`).
@jattasNI @rajsite @NIbokeefe @nate-ni @atmgrifter00 thoughts / votes?

Reply:
I vote fade-both. Meets the goal of not looking like motion while still changing state to look like something is progressing.

Reply:
Spec updated with a GIF of the prefers-reduced-motion animation that will be used (`fade-both` from the discussion above)